src/qibo/quantum_info/metrics.py

```diff
@@ -826,6 +858,39 @@ def expressibility(circuit, t: int, samples: int, backend=None):
         circuit, t, samples, backend=backend
     )
 
-    fid = fidelity(expr, expr)
+    fid = np.trace(expr @ expr)
 
     return fid
+
+
+def _check_hermitian_or_not_cupy(matrix: np.ndarray, backend=None):
+    """Checks if a given matrix is Hermitian and whether the backend is not :class:`qibojit.backends.CupyBackend`.
+
+    Args:
+        matrix (np.ndarray): input array.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses :class:`qibo.backends.GlobalBackend`.
+            Defaults to ``None``.
+
+    Returns:
+        bool: whether the matrix is Hermitian.
+
+    Raises:
+        NotImplementedError: If `matrix` is not Hermitian and `backend` is not :class:`qibojit.backends.CupyBackend`
+
+    """
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    hermitian = bool(
+        backend.calculate_norm(np.transpose(np.conj(matrix)) - matrix) < PRECISION_TOL
+    )
```

Comment:
```suggestion
    hermitian = (
        backend.calculate_norm(np.transpose(np.conj(matrix)) - matrix) < PRECISION_TOL
    )
```

Reply:
We need the `bool` function here otherwise the result of comparison is not actually type `bool`. You can test it yourself.